trino/client.py

```diff
@@ -607,7 +607,12 @@ def process(self, http_response) -> TrinoStatus:
             for name in get_header_values(
                 http_response.headers, constants.HEADER_DEALLOCATED_PREPARE
             ):
-                self._client_session.prepared_statements.pop(name)
+                # Multiple responses may contain the same deallocated statement name,
+                # so we can safely ignore `KeyError` if the same statement is already removed from the client session.
+                try:
+                    self._client_session.prepared_statements.pop(name)
+                except KeyError:
```

Comment:
avoid KeyError -- https://stackoverflow.com/a/15411146

Reply:
Done!

Reply:
Is there some way to know whether this results in a single lookup or two - multiple lookups will cause races when same connection has two cursors both of which are deallocating statements (highly unlikely since our statement names (and hence dict keys) are effectively unique).